### ISSUE TITLE
Samba stores wrong date on large files

### ISSUE THREAD

**Issue by AryehGielchinsky** [rclone/rclone#7411]
#### What is the problem you are having with rclone?
Rclone does not save the correct date using samba back-end for large files. This causes it to think it needs to recopy the file on subsequent syncs. Files below 300MB seem to work ok, while files above have the bug. Might be related to multi-thread copy.

#### What is your rclone version (output from `rclone version`)
```
rclone v1.64.2
- os/version: ubuntu 22.04 (64 bit)
- os/kernel: 5.15.0-86-generic (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.21.3
- go/linking: static
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Ubuntu 22.04 server 64 bit

#### Which cloud storage system are you using? (e.g. Google Drive)
Samba - ssd attached to my router running Openwrt connected by usb.

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
```rclone sync test/ router_smb:router_smb/samba_backup/test/ -vv```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
```
aryeh@server:~$ rclone sync test/ router_smb:router_smb/samba_backup/test/ -vv
2023/11/08 14:37:40 DEBUG : rclone: Version "v1.64.2" starting with parameters ["rclone" "sync" "test/" "router_smb:router_smb/samba_backup/test/" "-vv"]
2023/11/08 14:37:40 DEBUG : Creating backend with remote "test/"
2023/11/08 14:37:40 DEBUG : Using config file from "/home/aryeh/.config/rclone/rclone.conf"
2023/11/08 14:37:40 DEBUG : fs cache: renaming cache item "test/" to be canonical "/home/aryeh/test"
2023/11/08 14:37:40 DEBUG : Creating backend with remote "router_smb:router_smb/samba_backup/test/"
2023/11/08 14:37:41 DEBUG : fs cache: renaming cache item "router_smb:router_smb/samba_backup/test/" to be canonical "router_smb:router_smb/samba_backup/test"
2023/11/08 14:37:41 DEBUG : 2021-11-20 21.42.44.mp4: Need to transfer - File not found at Destination
2023/11/08 14:37:41 DEBUG : 2022-12-16 09.29.25.mp4: Need to transfer - File not found at Destination
2023/11/08 14:37:41 DEBUG : smb://smbuser@192.168.1.1:445/router_smb/samba_backup/test: Waiting for checks to finish
2023/11/08 14:37:41 DEBUG : smb://smbuser@192.168.1.1:445/router_smb/samba_backup/test: Waiting for transfers to finish
2023/11/08 14:37:41 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: write buffer set to 131072
2023/11/08 14:37:41 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: using backend concurrency of 4 instead of --multi-thread-streams 4
2023/11/08 14:37:41 DEBUG : 2022-12-16 09.29.25.mp4: Starting multi-thread copy with 5 chunks of size 64Mi with 4 parallel streams
2023/11/08 14:37:41 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 4/5 (201326592-268435456) size 64Mi starting
2023/11/08 14:37:41 DEBUG : 2022-12-16 09.29.25.mp4: writing chunk 3
2023/11/08 14:37:41 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 1/5 (0-67108864) size 64Mi starting
2023/11/08 14:37:41 DEBUG : 2022-12-16 09.29.25.mp4: writing chunk 0
2023/11/08 14:37:41 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 3/5 (134217728-201326592) size 64Mi starting
2023/11/08 14:37:41 DEBUG : 2022-12-16 09.29.25.mp4: writing chunk 2
2023/11/08 14:37:41 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 2/5 (67108864-134217728) size 64Mi starting
2023/11/08 14:37:41 DEBUG : 2022-12-16 09.29.25.mp4: writing chunk 1
2023/11/08 14:37:53 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 2/5 (67108864-134217728) size 64Mi finished
2023/11/08 14:37:53 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 5/5 (268435456-304863274) size 34.740Mi starting
2023/11/08 14:37:53 DEBUG : 2022-12-16 09.29.25.mp4: writing chunk 4
2023/11/08 14:37:53 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 3/5 (134217728-201326592) size 64Mi finished
2023/11/08 14:37:53 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 1/5 (0-67108864) size 64Mi finished
2023/11/08 14:37:53 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 4/5 (201326592-268435456) size 64Mi finished
2023/11/08 14:37:56 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 5/5 (268435456-304863274) size 34.740Mi finished
2023/11/08 14:37:56 DEBUG : 2022-12-16 09.29.25.mp4: Finished multi-thread copy with 5 parts of size 64Mi
2023/11/08 14:37:56 INFO  : 2022-12-16 09.29.25.mp4: Multi-thread Copied (new)
2023/11/08 14:37:58 INFO  : 2021-11-20 21.42.44.mp4: Copied (new)
2023/11/08 14:37:58 DEBUG : Waiting for deletions to finish
2023/11/08 14:37:58 INFO  : 
Transferred:      538.402 MiB / 538.402 MiB, 100%, 31.155 MiB/s, ETA 0s
Transferred:            2 / 2, 100%
Elapsed time:        17.5s

2023/11/08 14:37:58 DEBUG : 12 go routines active
2023/11/08 14:37:58 DEBUG : smb://smbuser@192.168.1.1:445/router_smb/samba_backup/test: Closing 2 unused connections
aryeh@server:~$ rclone ls router_smb:router_smb/samba_backup/test/ -vv
2023/11/08 14:38:06 DEBUG : rclone: Version "v1.64.2" starting with parameters ["rclone" "ls" "router_smb:router_smb/samba_backup/test/" "-vv"]
2023/11/08 14:38:06 DEBUG : Creating backend with remote "router_smb:router_smb/samba_backup/test/"
2023/11/08 14:38:06 DEBUG : Using config file from "/home/aryeh/.config/rclone/rclone.conf"
2023/11/08 14:38:06 DEBUG : smb://smbuser@192.168.1.1:445/router_smb/samba_backup/test: Using root directory "router_smb/samba_backup/test"
2023/11/08 14:38:06 DEBUG : fs cache: renaming cache item "router_smb:router_smb/samba_backup/test/" to be canonical "router_smb:router_smb/samba_backup/test"
259691740 2021-11-20 21.42.44.mp4
304863274 2022-12-16 09.29.25.mp4
2023/11/08 14:38:06 DEBUG : 7 go routines active
2023/11/08 14:38:06 DEBUG : smb://smbuser@192.168.1.1:445/router_smb/samba_backup/test: Closing 1 unused connections
aryeh@server:~$ rclone sync test/ router_smb:router_smb/samba_backup/test/ -vv
2023/11/08 14:38:12 DEBUG : rclone: Version "v1.64.2" starting with parameters ["rclone" "sync" "test/" "router_smb:router_smb/samba_backup/test/" "-vv"]
2023/11/08 14:38:12 DEBUG : Creating backend with remote "test/"
2023/11/08 14:38:12 DEBUG : Using config file from "/home/aryeh/.config/rclone/rclone.conf"
2023/11/08 14:38:12 DEBUG : fs cache: renaming cache item "test/" to be canonical "/home/aryeh/test"
2023/11/08 14:38:12 DEBUG : Creating backend with remote "router_smb:router_smb/samba_backup/test/"
2023/11/08 14:38:12 DEBUG : smb://smbuser@192.168.1.1:445/router_smb/samba_backup/test: Using root directory "router_smb/samba_backup/test"
2023/11/08 14:38:12 DEBUG : fs cache: renaming cache item "router_smb:router_smb/samba_backup/test/" to be canonical "router_smb:router_smb/samba_backup/test"
2023/11/08 14:38:12 DEBUG : smb://smbuser@192.168.1.1:445/router_smb/samba_backup/test: Waiting for checks to finish
2023/11/08 14:38:12 DEBUG : 2021-11-20 21.42.44.mp4: Size and modification time the same (differ by 0s, within tolerance 1ms)
2023/11/08 14:38:12 DEBUG : 2021-11-20 21.42.44.mp4: Unchanged skipping
2023/11/08 14:38:12 DEBUG : 2022-12-16 09.29.25.mp4: Modification times differ by 7853h8m31.5679419s: 2022-12-16 09:29:25 +0200 IST, 2023-11-08 14:37:56.5679419 +0200 IST
2023/11/08 14:38:12 DEBUG : smb://smbuser@192.168.1.1:445/router_smb/samba_backup/test: Waiting for transfers to finish
2023/11/08 14:38:12 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: write buffer set to 131072
2023/11/08 14:38:12 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: using backend concurrency of 4 instead of --multi-thread-streams 4
2023/11/08 14:38:12 DEBUG : 2022-12-16 09.29.25.mp4: Starting multi-thread copy with 5 chunks of size 64Mi with 4 parallel streams
2023/11/08 14:38:12 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 1/5 (0-67108864) size 64Mi starting
2023/11/08 14:38:12 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 4/5 (201326592-268435456) size 64Mi starting
2023/11/08 14:38:12 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 2/5 (67108864-134217728) size 64Mi starting
2023/11/08 14:38:12 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 3/5 (134217728-201326592) size 64Mi starting
2023/11/08 14:38:12 DEBUG : 2022-12-16 09.29.25.mp4: writing chunk 3
2023/11/08 14:38:12 DEBUG : 2022-12-16 09.29.25.mp4: writing chunk 1
2023/11/08 14:38:12 DEBUG : 2022-12-16 09.29.25.mp4: writing chunk 2
2023/11/08 14:38:12 DEBUG : 2022-12-16 09.29.25.mp4: writing chunk 0
2023/11/08 14:38:24 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 3/5 (134217728-201326592) size 64Mi finished
2023/11/08 14:38:24 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 5/5 (268435456-304863274) size 34.740Mi starting
2023/11/08 14:38:24 DEBUG : 2022-12-16 09.29.25.mp4: writing chunk 4
2023/11/08 14:38:24 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 2/5 (67108864-134217728) size 64Mi finished
2023/11/08 14:38:24 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 1/5 (0-67108864) size 64Mi finished
2023/11/08 14:38:24 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 4/5 (201326592-268435456) size 64Mi finished
2023/11/08 14:38:27 DEBUG : 2022-12-16 09.29.25.mp4: multi-thread copy: chunk 5/5 (268435456-304863274) size 34.740Mi finished
2023/11/08 14:38:27 DEBUG : 2022-12-16 09.29.25.mp4: Finished multi-thread copy with 5 parts of size 64Mi
2023/11/08 14:38:27 INFO  : 2022-12-16 09.29.25.mp4: Multi-thread Copied (replaced existing)
2023/11/08 14:38:27 DEBUG : Waiting for deletions to finish
2023/11/08 14:38:27 INFO  : 
Transferred:      290.740 MiB / 290.740 MiB, 100%, 20.472 MiB/s, ETA 0s
Checks:                 2 / 2, 100%
Transferred:            1 / 1, 100%
Elapsed time:        14.5s

2023/11/08 14:38:27 DEBUG : 11 go routines active
2023/11/08 14:38:27 DEBUG : smb://smbuser@192.168.1.1:445/router_smb/samba_backup/test: Closing 1 unused connections
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by AryehGielchinsky** [comment]
I've added ```--multi-thread-streams 0``` to the rclone command, and the date is copied correctly, so it appears to be an issue with the multi-thread copy on samba.

**Comment by ncw** [comment]
I've merged a fix for this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.65
